ui/src/components/Terminal/index.vue

```diff
@@ -0,0 +1,152 @@
+  (e: 'update:onlineUsers', onlineUsers: any[]): void;
+  (e: 'update:shareEnable', shareEnable: boolean): void;
+  (e: 'update:shareUserOptions', shareUserOptions: ShareUserOptions[]): void;
+}>();
+
+const { t } = useI18n();
+const message = useMessage();
+const terminalSettingsStore = useTerminalSettingsStore();
+const { connectionStatus, initializeSocketEvent, handleCreateShareUrl, getShareUser, handeleRemoveShareUser } =
+  useTerminalConnection(props.lunaId, props.origin);
+const { createTerminalInstance, terminalResizeEvent } = useTerminalInstance(props.socketInstance);
+
+onMounted(() => {
+  const { terminalId, enableShare, onlineUsers } = connectionStatus;
+
+  const terminalContainer: HTMLElement | null = document.getElementById('terminal-container');
+
+  if (!terminalContainer) {
+    return;
+  }
+
+  const terminalInstance: Terminal = createTerminalInstance(terminalContainer);
+
+  terminalInstance.open(terminalContainer);
+
+  if (!props.socketInstance) {
+    return;
+  }
+
+  initializeSocketEvent(terminalInstance, props.socketInstance, t);
+
+  mittBus.on('share-user', async ({ type, query }) => {
+    if (!props.socketInstance) {
+      return;
+    }
+
+    const userOptions: ShareUserOptions[] = await getShareUser(props.socketInstance, query);
+
+    emits('update:shareUserOptions', userOptions);
+  });
+  mittBus.on('create-share-url', async ({ type, shareLinkRequest }: { type: string; shareLinkRequest: any }) => {
+    if (!props.socketInstance) {
+      return;
+    }
+
+    const { shareId, shareCode } = await handleCreateShareUrl(props.socketInstance, shareLinkRequest);
+
+    emits('update:shareResult', {
+      shareId,
+      shareCode
+    });
+  });
+  mittBus.on('remove-share-user', ({ type, userMeta }) => {
+    if (!props.socketInstance) {
+      return;
+    }
+
+    handeleRemoveShareUser(props.socketInstance, userMeta);
+  });
+
+  watch(
+    () => enableShare.value,
+    newValue => {
+      emits('update:shareEnable', newValue);
+    }
+  );
+
+  watch(
+    () => terminalId.value,
+    id => {
+      if (id) {
+        terminalResizeEvent(terminalId.value);
+      }
+    }
+  );
+
+  watch(
+    () => terminalSettingsStore.theme,
+    theme => {
+      terminalInstance.options.theme = xtermTheme[theme!];
+
+      if (!props.socketInstance) {
+        return message.error('无法将主题同步到远端');
+      }
+
+      props.socketInstance?.send(
+        formatMessage(
+          terminalId.value,
+          'TERMINAL_SYNC_USER_PREFERENCE',
+          JSON.stringify({
+            terminal_theme_name: theme
+          })
+        )
+      );
+    }
+  );
+
+  watch(
+    () => onlineUsers.value,
+    userMap => {
+      emits('update:onlineUsers', userMap);
+    },
+    { deep: true }
+  );
+});
+</script>
+
+<style scoped lang="scss">
+#terminal-container {
+  :deep(.terminal) {
+    height: 100%;
+    padding: 10px 0 5px 10px;
+
+    .xterm-viewport {
+      overflow-y: unset !important;
+    }
+  }
+}
+</style>
```

Comment:
The code provided appears correct and there are no obvious irregularities or issues that I could detect.

Optimization Suggestions:

1. Remove unnecessary comments: Some of the commented-out lines can be removed to clean up and make the code more readable.

2. Simplify function names: Function names seem appropriate but if they're too generic, consider naming them more specifically for better readability.

3. Use proper indentation: Slight adjustments to indentations might improve legibility.

Overall, this is well structured code with minimal room for optimization changes without affecting its functionality significantly.